cpp/open3d/pipelines/integration/UniformTSDFVolume.cpp

```diff
@@ -257,28 +257,45 @@ std::shared_ptr<geometry::VoxelGrid> UniformTSDFVolume::ExtractVoxelGrid()
     voxel_grid->voxel_size_ = voxel_length_;
     voxel_grid->origin_ = origin_;
 
+    // Create a vector to hold voxels for each thread in the parallel region,
+    // since access to voxel_grid->voxels_ (std::unordered_map) is not
+    // thread-safe.
+    std::vector<std::vector<geometry::Voxel>> per_thread_voxels;
+
+#pragma omp parallel num_threads(utility::EstimateMaxThreads())
+    {
+#pragma omp single
+        { per_thread_voxels.resize(utility::GetNumThreads()); }
+        int thread_id = utility::GetThreadNum();
+
```

Comment:
`GetNumThreads()` is unnecessary. We can just use `EstimateMaxThreads` before the parallel region and use that to both set the size of per_thread_voxels and the parallelism.